connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/package.scala

```diff
@@ -17,5 +17,4 @@
 package org.apache.spark.sql
 
 package object protobuf {
```

Comment:
There are other empty `package object` definitions within the Spark internal code, such as 

https://github.com/apache/spark/blob/285489b0225004e918b6e937f7367e492292815e/streaming/src/main/scala/org/apache/spark/streaming/dstream/package.scala#L18-L24 

and 

https://github.com/apache/spark/blob/285489b0225004e918b6e937f7367e492292815e/core/src/main/scala/org/apache/spark/util/random/package.scala#L18-L23

Therefore, in this PR, only the definition of ·ScalaReflectionLock· was removed while the package object `protobuf` was retained.